backend/backend.go

```diff
@@ -99,6 +103,10 @@ type Operation struct {
 	// Input/output/control options.
 	UIIn  terraform.UIInput
 	UIOut terraform.UIOutput
+
+	// If LockState is true, the Operation must Lock any
+	// state.Lockers for its duration, and Unlock when complete.
```

Comment:
I don't think `backend.Local` is an exception here. The state is acquired and used solely within `Enhanced.Operation`, which is done for `backend.Local` too. I did note that `Backend.State` expects the caller to lock the state as needed, and that's called from within an Operation.